app/src/components/common/v2/Text.tsx

```diff
@@ -0,0 +1,152 @@
+import styled from '@emotion/styled';
+
+interface TextProps {
+  bold?: boolean;
+  semiBold?: boolean;
+  center?: boolean;
+  block?: boolean;
+  muted?: boolean;
+  space?: 4 | 8 | 12 | 16 | 20 | 24 | 32 | 40 | 48 | 56 | 64 | 96 | 120 | 160 | 200;
+  desktopSpace?:
+    | 4
+    | 8
+    | 12
+    | 16
+    | 20
+    | 24
+    | 32
+    | 40
+    | 48
+    | 56
+    | 64
+    | 96
+    | 120
+    | 160
+    | 200;
+}
+
+const BaseText = styled.span<TextProps>`
+  // On mobile, default to semi-bold when the bold prop is used
+  font-family: ${props =>
+    props.bold || props.semiBold
+      ? props.theme.fonts.open.semiBold
+      : props.theme.fonts.open.regular};
+  font-weight: ${props => (props.bold || props.semiBold ? 600 : 400)};
+  ${props => props.muted && `color: ${props.theme.colors.gray};`}
```

Comment:
Should we stay consistent with
`color: ${props => (props.muted && 'props.theme.colors.gray')}`

Reply:
The difference is that the font weight always has a value. If the muted property doesn't exist there would be no color in the code you suggested.